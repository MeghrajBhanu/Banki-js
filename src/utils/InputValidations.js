export const validateEmail = (checkingText) => {
    
    const regexp = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/; 
   
    
    if (regexp.exec(checkingText) !== null) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Email Address'
            };
        }
}
export const validatePan = (checkingText) => {
    
    
    if (checkingText.length === 10) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Pan Number'
            };
        }
}
export const validatePass = (checkingText) => {
    
    
    if (checkingText!=="") {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Password'
            };
        }
}