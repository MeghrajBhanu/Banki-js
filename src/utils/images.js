import bank1 from "../assets/bank1.jpg";
import bank2 from "../assets/bank2.png";
import bank3 from "../assets/bank3.jfif";
import bank4 from "../assets/bank4.png";

const image_map={
    "Bank1":bank1,
    "Bank2":bank2,
    "Bank3":bank3,
    "Bank4":bank4,
}
export default image_map;