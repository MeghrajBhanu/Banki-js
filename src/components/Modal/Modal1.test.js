import { render, screen, waitFor } from "@testing-library/react";
import { onlywrap, renderComponent } from "../../utils/test_helper";
import renderer from "react-test-renderer";
import Modal1 from "./Modal1";

describe("Modal1", () => {
  test("snapshot test", () => {
    const tree = renderer.create(onlywrap(<Modal1 />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});