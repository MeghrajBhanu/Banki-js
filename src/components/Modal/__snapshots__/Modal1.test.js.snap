// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal1 snapshot test 1`] = `
<div
  className="py-5 my-5 container-fluid"
>
  <div
    className="row px-0 mt-3 align-content-center"
  >
    <div
      className="card mx-auto"
      style={
        Object {
          "width": "25rem",
        }
      }
    >
      <h4
        className="card-header text-center"
      >
        <h6>
          App
        </h6>
      </h4>
      <i
        className="bi bi-check-circle-fill text-success"
        style={
          Object {
            "fontSize": "50px",
            "textAlign": "center",
          }
        }
      />
      <div
        className="card-body"
      >
        <h4
          className="text-center"
        >
          Successfull!
        </h4>
        <h4
          className="text-center"
        >
          Dear User,
        </h4>
        <h4
          className="card-text text-center"
        >
          Your Account has been unflagged!
        </h4>
        <div
          className="mb-0 text-center"
        >
          <p
            className="mb-0"
          >
            Request to your bank for a revalidation has been cancelled.
          </p>
        </div>
      </div>
      <div
        className="card-footer d-flex"
      >
        <button
          className="btn btn-info rounded-pill w-100 ms-2 "
          onClick={[Function]}
        >
          Return
        </button>
      </div>
    </div>
  </div>
</div>
`;
